packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -3846,6 +3846,468 @@ class SliverPadding extends SingleChildRenderObjectWidget {
   }
 }
 
+/// A widget that annotates the widget tree with a description of the meaning of
+/// the widgets.
+///
+/// {@template flutter.widgets.SemanticsBase}
+/// Used by assistive technologies, search engines, and other semantic analysis
+/// software to determine the meaning of the application.
+///
+/// See also:
+///
+///  * [SemanticsProperties], which contains a complete documentation for each
+///    of the constructor parameters that belongs to semantics properties.
+///  * [MergeSemantics], which marks a subtree as being a single node for
+///    accessibility purposes.
+///  * [ExcludeSemantics], which excludes a subtree from the semantics tree
+///    (which might be useful if it is, e.g., totally decorative and not
+///    important to the user).
+///  * [RenderObject.describeSemanticsConfiguration], the rendering library API
+///    through which the [Semantics] widget is actually implemented.
+///  * [SemanticsNode], the object used by the rendering library to represent
+///    semantics in the semantics tree.
+///  * [SemanticsDebugger], an overlay to help visualize the semantics tree. Can
+///    be enabled using [WidgetsApp.showSemanticsDebugger],
+///    [MaterialApp.showSemanticsDebugger], or [CupertinoApp.showSemanticsDebugger].
+/// {@endtemplate}
+sealed class SemanticsBase extends SingleChildRenderObjectWidget {
```

Comment:
Thanks for refactor this out it is a lot less duplicated code now, I wonder if this can be private? not sure whether the api doc still work if we make it private though.

Reply:
Confirmed the API docs still work even if we make it private.